### PR TITLE
[backend/frontend] Introduce dev flags to deactivate feature in ongoing development

### PR DESCRIPTION
**Pull request by richard-julien**
This PR aims to introduce a dev configuration to disable features that are currently in development but not ready yet.
That's will ease the merging of large feature but providing simple tools to hide the feature.

In this PR you will find a example of Decay indicator feature deactivation.
- Deactivation of the manager registration
- Deactivation of the module registration
- Deactivation of a frontend specific menu

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/settings/CustomizationMenu.tsx [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import React, { FunctionComponent } from 'react';
 import NavToolbarMenu, { MenuEntry } from '../common/menus/NavToolbarMenu';
+import useHelper from '../../../utils/hooks/useHelper';
 
 const CustomizationMenu: FunctionComponent = () => {
+  const { isFeatureEnable } = useHelper();
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/private/components/settings/CustomizationMenu.tsx]:
```suggestion
  const { isFeatureEnabled } = useHelper();
```

File: opencti-platform/opencti-front/src/private/components/settings/CustomizationMenu.tsx [hunk 2]
```diff
@@ -19,6 +21,10 @@ const CustomizationMenu: FunctionComponent = () => {
       path: '/dashboard/settings/customization/retention',
       label: 'Retention policies',
     },
+    ...(isFeatureEnable('INDICATOR_DECAY') ? [{
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/private/components/settings/CustomizationMenu.tsx]:
```suggestion
    ...(isFeatureEnabled('INDICATOR_DECAY') ? [{
```

File: opencti-platform/opencti-front/src/utils/platformModulesHelper.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ const isFeatureEnable = (
 ) => {
   const flags = settings.platform_feature_flags ?? [];
   const feature = flags.find((f) => f.id === id);
-  return feature !== undefined && feature.enable === true;
+  return feature === undefined || feature.enable === true;
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/utils/platformModulesHelper.ts:39]:
```suggestion
  return feature === undefined || feature.enabled === true;
```

File: opencti-platform/opencti-front/src/private/components/settings/Settings.jsx [hunk 4]
```diff
@@ -972,9 +971,7 @@ const Settings = () => {
                               <ListItemText primary={t_i18n(module.id)} />
                               <ItemBoolean
                                 variant="inList"
-                                label={
-                                  module.enable ? t_i18n('Enabled') : t_i18n('Disabled')
-                                }
+                                label={module.enable ? t_i18n('Enabled') : t_i18n('Disabled')}
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/private/components/settings/Settings.jsx:974]:
```suggestion
                                label={module.enabled ? t_i18n('Enabled') : t_i18n('Disabled')}
```

**Review by aHenryJard — COMMENTED** [review]
I just did a PR review, I did not tried it yet.

To avoid mispelling, I think it's better to have the 'ed' everywhere (enabled, disabled, activated etc..) so I did suggestions in that way.

**Comment by labo-flg** [comment]
Let's enable all feature flags in the backend tests, to cover the code during development even if not enabled in production yet.

**Comment by richard-julien** [comment]
> I just did a PR review, I did not tried it yet.
> 
> To avoid mispelling, I think it's better to have the 'ed' everywhere (enabled, disabled, activated etc..) so I did suggestions in that way.

That's a good point but there is more work and alignment to do than your proposal to really align everything around "enabled" like the modules that share the graphql object for the result. Modules "enable" is currently written in redis so the modification of the name must be changed carefully.

For now i will propose to only focus on the flag here and do the change of semantic in another PR

**Review by aHenryJard — APPROVED** [review]
Tested locally on Decay feature, it's working as expected.

I tested:
- without disabled_dev_features => decay hidden, ok
- with disabled_dev_features empty => decay enabled, ok
- with disabled_dev_features = ['INDICATOR_DECAY']  => decay hidden, ok

**Comment by richard-julien** [comment]
> Let's enable all feature flags in the backend tests, to cover the code during development even if not enabled in production yet.

Done

**Review by labo-flg — APPROVED** [review]

**Pull request merged**